### PR TITLE
fix: notification template dual state

### LINKED ISSUE #3386: [NV-2292] 🐛 Bug Report: dual active states in UI, if template is created with api
### 📜 Description

When I create a Novu Notification Template using the Novu Node SDK the UI shows that it is in both active and disabled state. I can't trigger notifications using this template until I toggle the active state off and back on in the UI.
err

### 👟 Reproduction steps

1. Use the Novu Node SDK to create a notification group and then a notification template with the same name.
2. Get the triggerID for this notification template and trigger a notification to subscribers (I used 1) with this template.
3. Go to UI and see no notification in the activity and the notification template is in a dual active/disabled state.

### 👍 Expected behavior

It should create the Notification Template in active state and allow me to trigger notifications using this template.

### 👎 Actual Behavior with Screenshots

Screenshots attached in description

### 📃 Provide any additional context for the Bug.

_No response_

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Contributing Guidelines?

- [X] I have read the [Contributing Guidelines](https://github.com/novuhq/novu/blob/main/CONTRIBUTING.md)

### Are you willing to submit PR?

None

<sub>[NV-2292](https://linear.app/novu/issue/NV-2292/🐛-bug-report-dual-active-states-in-ui-if-template-is-created-with-api)</sub>

### PR DESCRIPTION
**Pull request by michaldziuba03**
### What change does this PR introduce?
It fixes bug where it was possible to create notification template with both `enabled` and `disabled` states at the same time.

screenshot

Something like this is only possible when template is created using API or SDK.

Bugs happen when `active` and `draft` are equal, so I decided to deprecate `draft` field in DTO. After my changes `draft` is always set as opposite of `active` value provided by the user (this is exactly what frontend does but SDK users can set `draft` value however they want). I also fixed this problem on frontend side where I removed `draft` from render conditions.

### Why was this change needed?
Closes #3386

### CODE REVIEW

File: apps/api/src/app/events/usecases/parse-event-request/parse-event-request.usecase.ts [hunk 1]
```diff
@@ -3,7 +3,7 @@ import * as Sentry from '@sentry/node';
 import * as hat from 'hat';
 import { merge } from 'lodash';
 import { v4 as uuidv4 } from 'uuid';
-import { AnalyticsService, Instrument, InstrumentUsecase } from '@novu/application-generic';
```
Comment (michaldziuba03) [inline apps/api/src/app/events/usecases/parse-event-request/parse-event-request.usecase.ts:6]: some unused service

File: apps/api/src/app/notification-template/dto/create-notification-template.request.dto.ts [hunk 2]
```diff
@@ -42,7 +42,7 @@ export class CreateNotificationTemplateRequestDto implements ICreateNotification
   @IsOptional()
   active?: boolean;
 
-  @ApiPropertyOptional()
+  @ApiPropertyOptional({ deprecated: true })
```
Comment (michaldziuba03) [inline apps/api/src/app/notification-template/dto/create-notification-template.request.dto.ts:45]: mark as deprecated in swagger

File: apps/api/src/app/notification-template/notification-template.controller.ts [hunk 3]
```diff
@@ -174,7 +174,7 @@ export class NotificationTemplateController {
         steps: body.steps,
         notificationGroupId: body.notificationGroupId,
         active: body.active ?? false,
-        draft: body.draft ?? true,
+        draft: !body.active,
```
Comment (michaldziuba03) [inline apps/api/src/app/notification-template/notification-template.controller.ts:177]: always set `draft` as opposite of `value`. Ignore deprecated `body.draft`.

This is exactly what frontend does - but SDK users can set draft/active however they want what may lead them to bugs.

File: apps/api/src/app/notification-template/usecases/change-template-active-status/change-template-active-status.usecase.ts [hunk 4]
```diff
@@ -55,7 +55,7 @@ export class ChangeTemplateActiveStatus {
       {
         $set: {
           active: command.active,
-          draft: command.active === false,
+          draft: !command.active,
```
Comment (michaldziuba03) [inline apps/api/src/app/notification-template/usecases/change-template-active-status/change-template-active-status.usecase.ts:58]: just shorter version suggested by editor

File: apps/api/src/app/events/usecases/parse-event-request/parse-event-request.usecase.ts [hunk 5]
```diff
@@ -56,7 +55,7 @@ export class ParseEventRequest {
       throw new UnprocessableEntityException('workflow_not_found');
     }
 
-    if (!template.active || template.draft) {
```
Comment (michaldziuba03) [inline apps/api/src/app/events/usecases/parse-event-request/parse-event-request.usecase.ts:59]: after change users may still have those "broken" templates with active=true and draft=true.

UI after my changes will show that template is active, but API will be unable to trigger event becasue `draft` is true.

File: apps/web/src/pages/templates/WorkflowListPage.tsx [hunk 6]
```diff
@@ -71,7 +71,7 @@ const columns: IExtendedColumn<INotificationTemplateExtended>[] = [
     maxWidth: 125,
     Cell: withCellLoading(({ row: { original } }) => (
       <>
-        {original.draft ? (
+        {!original.active ? (
```
Comment (michaldziuba03) [inline apps/web/src/pages/templates/WorkflowListPage.tsx:74]: after change users may still have those "broken" templates with active=true and draft=true.

so without this change on frontned they could still see dual state :)

File: apps/api/src/app/notification-template/dto/update-notification-template-request.dto.ts [hunk 7]
```diff
@@ -37,9 +37,6 @@ export class UpdateNotificationTemplateRequestDto implements ICreateNotification
   @IsMongoId()
   notificationGroupId: string;
 
-  @ApiPropertyOptional()
-  active?: boolean;
```
Comment (michaldziuba03) [inline apps/api/src/app/notification-template/dto/update-notification-template-request.dto.ts:41]: this property from DTO is not even passed to the usecase in controller.

Application has seperated endpoint to update status

**Review by scopsy — APPROVED** [review]
Fantastic work @michaldziuba03!

**Pull request merged**